ColorSwitcher/Model/GIFManager.swift

```
@@ -0,0 +1,58 @@
+//
+//  GIFManager.swift
+//  ColorSwitcher
+//
+//  Created by Matt M Smith on 9/28/20.
+//
+
+import Foundation
+
+protocol GIFManagerDelegate {
+    func getGIFModel(title: String, urlString: String)
+}
+
+struct GIFManager {
+    var delegate: GIFManagerDelegate?
+    
+    private let apiKey = "0OZKMCWSOD4M"
+    private let baseURL = "https://api.tenor.com/v1/search?"
+    
+    func fetchGIF(from query: String) {
+        let formattedQuery = query.replacingOccurrences(of: " ", with: "+")
+        let urlString = "\(baseURL)?api_key=\(apiKey)&q=\(formattedQuery)&contentfilter=high&media_filter=minimal&limit=1"
+        
+        if let url = URL(string: urlString) {
+            let session = URLSession(configuration: .default)
+            let task = session.dataTask(with: url) { (data, _, error) in
+                if let error = error {
+                    print(error)
```

Failing silently deliberately here?